kernel/src/paging.c

```diff
@@ -148,7 +233,7 @@ void rec_setpage(PageEntry* table, uint64_t table_addr, uint8_t depth, MapParams
 
 void map_pages(PageEntry* table, uint64_t low, uint64_t high, uint64_t phys_low, PageFlags flags) {
     MapParams params = {
-        low & PAGING_PTRMASK, high & PAGING_PTRMASK, (phys_low - low) & PAGING_PTRMASK, flags};
```

Comment:
Why can we just remove the subtraction here?

Reply:
But why can't this underflow?

Reply:
since all valid mappings will satisfy 
`table_addr + params->physaddr_low >= params->virtaddr_low`

Reply:
Could you link me a source, I believe you I just to see where you found the information.

Reply:
table_addr is in the bounds of [viraddrs_low, viraddrs_high]. So it's natural that it's bigger or equal to virtaddrs_low.
I will however move it down to set_mapping, since it's the only place it's used.

Reply:
What's the range of viraddr?

Reply:
Because I'm not so sure that this can't underflow. You can map high virtual addresses to low physical addresses. Why can't this happen?

Reply:
misspelled virtaddr previously.

table_addr is moved to virtaddrs_low at the start of the rec_setmap function, as described here
  
```c
// align table address with virutal address
// this is the same as finding the lower bound of a table
if (table_addr < params->virtaddr_low) {
    table_addr += params->virtaddr_low & pagingmasks[depth];
}
```
table_addr >= virtaddrs_low
therefore, 
table_addr + (positive number) - virtaddrs_low >= 0 

no underflow, QED.

Reply:
That's what I wanted to see, it was really hard to see where the change that made sure of this occurred because there's only one big commit. In future try to create multiple commits with more contained changes, it'll make the life of the reviewer much easier and create less confusion. You can still create one PR but create multiple commits which describe what changes have been made and why.